src/cli.rs

```diff
@@ -145,6 +152,65 @@ fn time_details(
     )
 }
 
+/// Structure representing a series of channel ranges
+/// e.g. 1-10, 20-30, 40-50
+#[derive(Debug, Clone)]
+pub struct ChannelRanges {
+    /// Vector of channel ranges
+    pub ranges: Vec<(usize, usize)>,
+}
+
+impl ChannelRanges {
+    /// Create a new ChannelRanges object from a string
+    /// e.g. "1-10, 20-30, 40-50"
+    pub fn new(s: &str) -> Result<Self, BirliError> {
+        let mut ranges = Vec::new();
+        for range in s.split(',') {
+            match range.split('-').collect::<Vec<_>>().as_slice() {
+                [start, end] => {
+                    let start_result = start.trim().parse::<usize>();
+                    let end_result = end.trim().parse::<usize>();
+                    match (start_result, end_result) {
+                        (Ok(start), Ok(end)) => {
+                            ranges.push((start, end));
+                        }
+                        _ => {
+                            return Err(BirliError::CLIError(InvalidRangeSpecifier {
+                                reason: format!("invalid channel range: {}", range),
+                            }));
+                        }
+                    }
+                }
+                _ => {
+                    return Err(BirliError::CLIError(InvalidRangeSpecifier {
+                        reason: format!("invalid channel range: {}", range),
+                    }));
+                }
+            }
+        }
+        Ok(ChannelRanges { ranges })
+    }
+
+    /// Create a new ChannelRanges object spanning all available channels in the metafits context
+    pub fn all(context: &CorrelatorContext) -> Self {
+        let mut coarse_chan_indices = context.common_coarse_chan_indices.iter();
+        let mut ranges: Vec<(usize, usize)> = Vec::new();
+        let mut range_start = *coarse_chan_indices.next().unwrap();
```

Comment:
Can the unwrap be avoided? I think there's a `start` or `first` for range objects?